src/app/parcels/batch/BatchParcelsReducer.ts

```diff
@@ -0,0 +1,282 @@
+import { BooleanGroup } from "@/components/DataInput/inputHandlerFactories";
+import { Person } from "@/components/Form/formFunctions";
+import { logErrorReturnLogId } from "@/logger/logger";
+import { getClientFromClients, getAllPeopleFromFamily } from "@/app/parcels/batch/supabaseHelpers";
+import { NAPPY_SIZE_LABEL, EXTRA_INFORMATION_LABEL } from "@/app/clients/form/labels";
+import {
+    BatchTableDataState,
+    BatchActionType,
+    BatchClient,
+    BatchDataRow,
+    OverrideClient,
+    OverrideParcel,
+    clientOverrideCellValueType,
+    parcelOverrideCellValueType,
+    BatchParcel,
+} from "@/app/parcels/batch/BatchTypes";
+import dayjs from "dayjs";
+
+const getOverridenFieldsAndValues = (
+    allFields: OverrideClient | OverrideParcel
+): (
+    | { field: string; value: clientOverrideCellValueType }
+    | { field: string; value: parcelOverrideCellValueType }
+)[] => {
+    return Object.entries(allFields)
+        .filter(([_, value]) => value)
+        .reduce(
+            (acc, [key, value]) => {
+                return [...acc, { field: key, value: value }];
+            },
+            [] as (
+                | { field: string; value: clientOverrideCellValueType }
+                | { field: string; value: parcelOverrideCellValueType }
+            )[]
+        );
+};
+
+function createBooleanGroupFromStrings(strings: string[] | null): BooleanGroup {
+    const result: BooleanGroup = {};
+    if (strings) {
+        strings.forEach((str) => {
+            result[str] = true;
+        });
+    }
+    return result;
+}
+
+const getNappySize = (info: string | null): string | null => {
+    if (info) {
+        const match = info.match(new RegExp(`${NAPPY_SIZE_LABEL}(\\d+)`));
+        if (match) {
+            return match[1];
+        }
+    }
+    return null;
+};
+
+const getChildrenAndAdults = async (
+    familyId: string
+): Promise<{ adults: Person[]; children: Person[] } | null> => {
+    const { data, error } = await getAllPeopleFromFamily(familyId);
+
+    const adults: Person[] = [];
+    const children: Person[] = [];
+
+    //do not run this test at 23:59:59.99 on the last day of the month
+    //this will cause the test to fail
```

Comment:
This isn't in a test file? If we need to have this comment can we move it to the test file